app/controllers/AiModelController.scala

```diff
@@ -0,0 +1,224 @@
+}
+
+case class UpdateAiModelParameters(name: String, comment: Option[String])
+
+object UpdateAiModelParameters {
+  implicit val jsonFormat: OFormat[UpdateAiModelParameters] = Json.format[UpdateAiModelParameters]
+}
+
+class AiModelController @Inject()(
+    aiModelDAO: AiModelDAO,
+    aiModelService: AiModelService,
+    sil: Silhouette[WkEnv],
+    userService: UserService,
+    annotationDAO: AnnotationDAO,
+    aiInferenceService: AiInferenceService,
+    aiInferenceDAO: AiInferenceDAO,
+    organizationDAO: OrganizationDAO,
+    datasetService: DatasetService,
+    jobService: JobService,
+    datasetDAO: DatasetDAO,
+    dataStoreDAO: DataStoreDAO)(implicit ec: ExecutionContext, bodyParsers: PlayBodyParsers)
+    extends Controller
+    with FoxImplicits {
+
+  def readAiModelInfo(aiModelId: String): Action[AnyContent] = sil.SecuredAction.async { implicit request =>
+    {
+      for {
+        _ <- userService.assertIsSuperUser(request.identity)
+        aiModelIdValidated <- ObjectId.fromString(aiModelId)
+        aiModel <- aiModelDAO.findOne(aiModelIdValidated) ?~> "aiModel.notFound" ~> NOT_FOUND
+        jsResult <- aiModelService.publicWrites(aiModel)
+      } yield Ok(jsResult)
+    }
+  }
+
+  def readAiInferenceInfo(aiInferenceId: String): Action[AnyContent] = sil.SecuredAction.async { implicit request =>
+    {
+      for {
+        _ <- userService.assertIsSuperUser(request.identity)
+        aiInferenceIdValidated <- ObjectId.fromString(aiInferenceId)
+        aiInference <- aiInferenceDAO.findOne(aiInferenceIdValidated) ?~> "aiInference.notFound" ~> NOT_FOUND
+        jsResult <- aiInferenceService.publicWrites(aiInference, request.identity)
+      } yield Ok(jsResult)
+    }
+  }
+
+  def listAiModels: Action[AnyContent] = sil.SecuredAction.async { implicit request =>
+    {
+      for {
+        _ <- userService.assertIsSuperUser(request.identity)
+        aiModels <- aiModelDAO.findAll
+        jsResults <- Fox.serialCombined(aiModels)(aiModelService.publicWrites)
+      } yield Ok(Json.toJson(jsResults))
+    }
+  }
+
+  def listAiInferences: Action[AnyContent] = sil.SecuredAction.async { implicit request =>
+    {
+      for {
+        _ <- userService.assertIsSuperUser(request.identity)
+        aiInferences <- aiInferenceDAO.findAll
+        jsResults <- Fox.serialCombined(aiInferences)(inference =>
+          aiInferenceService.publicWrites(inference, request.identity))
+      } yield Ok(Json.toJson(jsResults))
+    }
+  }
+
+  def runTraining: Action[RunTrainingParameters] = sil.SecuredAction.async(validateJson[RunTrainingParameters]) {
+    implicit request =>
+      for {
+        _ <- userService.assertIsSuperUser(request.identity)
+        trainingAnnotations = request.body.trainingAnnotations
+        _ <- bool2Fox(trainingAnnotations.nonEmpty || request.body.workflowYaml.isDefined) ?~> "aiModel.training.zeroAnnotations"
```

Comment:
No error message exists for `"aiModel.training.zeroAnnotations"` in messages.conf